openedx_rocketchat/static/js/rocket_chat_discussion.js

```diff
@@ -0,0 +1,75 @@
+
+class RocketChatNavigation {
+
+    constructor(url_change_role) {
+        this.pageContentMain = $(".page-content-main");
+        this.pageContentSecondary = $(".page-content-secondary");
+        this.url_change_role = url_change_role;
+
+        this.loadComponents(this.pageContentMain, this.pageContentSecondary);
+
+        $(window).bind('hashchange', $.proxy(function(){
+            this.loadComponents(this.pageContentMain, this.pageContentSecondary);
+        },this));
+
+        $(".change-role-rocketchat").unbind().click($.proxy(function() {
+            this.change_role(this.url_change_role);
+        }, this))
+    }
+
+
+    change_role(url_change_role){
+        var input = $(".rocket-chat-user-input")
+        var username = input.val();
+        var role = "coach";
+        $.ajax({
+            type: "POST",
+            url: url_change_role,
+            data: {"username": username, "role": role},
+            success: function(){
```

Comment:
Please, try to handle each status code that the API can response, to ensure that we have a correct implementation for every staus code. @andrey-canon